clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -220,6 +225,95 @@ public void testAssignOnEmptyTopicInPartition() {
+    }
+
+    @Test
+    public void testBeginningOffsetsTimeoutOnEventProcessingTimeout() {
+        PrototypeAsyncConsumer<?, ?> consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
+        doThrow(new TimeoutException()).when(eventHandler).addAndGet(any(), any());
+        assertThrows(TimeoutException.class,
+                () -> consumer.beginningOffsets(
+                        Collections.singletonList(new TopicPartition("t1", 0)),
+                        Duration.ofMillis(1)));
+        verify(eventHandler).addAndGet(ArgumentMatchers.isA(ListOffsetsApplicationEvent.class),
+                ArgumentMatchers.isA(Timer.class));
+    }
+
+    @Test
+    public void testOffsetsForTimesOnNullPartitions() {
+        PrototypeAsyncConsumer<?, ?> consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
+        assertThrows(NullPointerException.class, () -> consumer.offsetsForTimes(null,
+                Duration.ofMillis(1)));
+    }
+
+    @Test
+    public void testOffsetsForTimes() {
+        PrototypeAsyncConsumer<?, ?> consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
+        Map<TopicPartition, OffsetAndTimestamp> expectedResult = mockOffsetAndTimestamp();
+        Map<TopicPartition, Long> timestampToSearch = mockTimestampToSearch();
+
+        doReturn(expectedResult).when(eventHandler).addAndGet(any(), any());
+        Map<TopicPartition, OffsetAndTimestamp> result =
+                assertDoesNotThrow(() -> consumer.offsetsForTimes(timestampToSearch, Duration.ofMillis(1)));
+        assertEquals(expectedResult, result);
+        verify(eventHandler).addAndGet(ArgumentMatchers.isA(ListOffsetsApplicationEvent.class),
+                ArgumentMatchers.isA(Timer.class));
+    }
+
+    // This test ensures same behaviour as the current consumer when offsetsForTimes is called
+    // with 0 timeout. It should return map with all requested partitions as keys, with null
+    // OffsetAndTimestamp as value.
+    @Test
+    public void testOffsetsForTimesWithZeroTimeout() {
+        PrototypeAsyncConsumer<?, ?> consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
+        TopicPartition tp = new TopicPartition("topic1", 0);
+        Map<TopicPartition, OffsetAndTimestamp> expectedResult =
+                Collections.singletonMap(tp, null);
+        Map<TopicPartition, Long> timestampToSearch = Collections.singletonMap(tp, 5L);
+
+        Map<TopicPartition, OffsetAndTimestamp> result =
+                assertDoesNotThrow(() -> consumer.offsetsForTimes(timestampToSearch,
+                        Duration.ofMillis(0)));
+        assertEquals(expectedResult, result);
+        verify(eventHandler, never()).addAndGet(ArgumentMatchers.isA(ListOffsetsApplicationEvent.class),
+                ArgumentMatchers.isA(Timer.class));
+    }
```

Comment:
This is an existing issue. `testWakeup_commitSync` and `clientId` seem unused?

Reply:
Removed the unused code. 

Do you refer to a reported issue with the behaviour of the consumer `offsetsForTimes` returning null when 0 timeout? (can't find any). 

For now we want to ensure the same API level behaviour so we are including this test, similar to the one defined for the old Consumer [here](https://github.com/apache/kafka/blob/6391c6603582a4e4b5bb670233c345e53f82a77b/clients/src/test/java/org/apache/kafka/clients/consumer/internals/OffsetFetcherTest.java#L971). Both wil definitely have to be updated/removed eventually if we change the behaviour.